tesseract_common/include/tesseract_common/resource_locator.h

```diff
@@ -82,7 +84,18 @@ class GeneralResourceLocator : public ResourceLocator
 public:
   using Ptr = std::shared_ptr<GeneralResourceLocator>;
   using ConstPtr = std::shared_ptr<const GeneralResourceLocator>;
+  /**
+   * @brief Construct a new General Resource Locator object using the TESSERACT_RESOURCE_PATH environment variable
+   */
   GeneralResourceLocator();
+  /**
+   * @brief Construct a new General Resource Locator object using the provided paths and/or the TESSERACT_RESOURCE_PATH
+   * environment variable
+   *
+   * @param paths A vector of paths to search for resources
+   * @param use_environment If true, the TESSERACT_RESOURCE_PATH environment variable will also be used
+   */
+  GeneralResourceLocator(const std::vector<tesseract_common::fs::path>& paths, bool use_environment = true);
```

Comment:
Is the use_environment needed? If they do not what to use the environment variable shouldn't they not defined the environment variable?

Reply:
The system or ROS may set these variables automatically. It is better to have the option to opt in or out of the env variable.

Reply:
I have made the suggested changes

Reply:
I guess, why we are at should we just expose a loadEnvironmentVariable function?

Reply:
I renamed and made the function public